master/internal/templates/api.go

```diff
@@ -109,29 +109,56 @@ func (a *TemplateAPIServer) PutTemplate(
 	if len(req.Template.Name) == 0 {
 		return nil, status.Error(codes.InvalidArgument, "name is required")
 	}
-	if req.Template.WorkspaceId != 0 {
-		return nil, status.Errorf(codes.InvalidArgument, "setting workspace_id is not supported.")
-	}
 
-	switch _, err := TemplateByName(ctx, req.Template.Name); {
+	switch tpl, err := TemplateByName(ctx, req.Template.Name); {
 	case errors.Is(err, db.ErrNotFound):
 		_, err := a.PostTemplate(ctx, &apiv1.PostTemplateRequest{Template: req.Template})
 		if err != nil {
 			return nil, err
 		}
+		return &apiv1.PutTemplateResponse{Template: req.Template}, nil
 	case err != nil:
 		return nil, err
 	default:
-		req := &apiv1.PatchTemplateConfigRequest{
-			TemplateName: req.Template.Name,
-			Config:       req.Template.Config,
+		user, _, err := grpcutil.GetUser(ctx)
```

Comment:
nit: Since there's more complexity in the default case, and nothing else happens in the function, I would prefer simplifying the default case a bit. Either leave off `default:` and let everything that happens in `default:` happen as part of normal function execution. Or, if it makes sense as a logical unit, move some of it into separate functions.

Reply:
Sure